src/App.tsx

```diff
@@ -15,33 +15,133 @@ const goodsFromServer: string[] = [
   'Garlic',
 ];
 
-export const App: React.FC = () => (
-  <div className="App">
-    <button type="button">
-      Start
-    </button>
-
-    <button type="button">
-      Sort alphabetically
-    </button>
-
-    <button type="button">
-      Sort by length
-    </button>
-
-    <button type="button">
-      Reverse
-    </button>
-
-    <button type="button">
-      Reset
-    </button>
-
-    <ul className="Goods">
-      <li className="Goods__item">Dumplings</li>
-      <li className="Goods__item">Carrot</li>
-      <li className="Goods__item">Eggs</li>
-      <li className="Goods__item">...</li>
-    </ul>
-  </div>
-);
+enum SortType {
+  NONE = 'NONE',
+  ALPHABET = 'ALPHABET',
+  LENGTH = 'LENGTH',
+}
+
+function getReorderedGoods(
+  goods: string[],
+  sortBy: SortType,
+  reversedGoods: boolean,
+): string[] {
+  const showedGoods = [...goods];
+
+  switch (sortBy) {
+    case SortType.ALPHABET:
+      showedGoods.sort(
+        (currentProduct, nextProduct) => (
+          currentProduct.localeCompare(nextProduct)
+        ),
+      );
+      break;
+
+    case SortType.LENGTH:
+      showedGoods.sort(
+        (currentProduct, nextProduct) => (
+          currentProduct.length - nextProduct.length
+        ),
+      );
+      break;
+
+    default:
+      break;
+  }
+
+  if (reversedGoods) {
+    return showedGoods.reverse();
+  }
+
+  return showedGoods;
+}
+
+export const App: React.FC = () => {
+  const [isStarted, setStarted] = useState(false);
+  const [isReversed, setReversed] = useState(false);
+  const [sortType, setSortType] = useState(SortType.NONE);
+
+  function handleStart() {
```

Comment:
arguable but still a suggestion: named function is not a common practice, arrow function is your choice (